routes/article.js

```diff
@@ -138,29 +136,54 @@ router.post('/vote', (req, res) => {
             })
             console.log(`article: ${articleChanged}`)
             res.send(articleChanged)
-            return
-
         })
-        .catch((error) => {
+        .catch((err) => {
             res.status(500).send({
-                error,
+                err,
                 reqBody: req.body
             })
-            return
         })
 })
 
-router.get('/', (req, res) => {
-    Article
-        .find()
-        .lean()
-        .then((articles) => {
-            res.send(articles) 
+router.get('/top', (req, res) => {
+    let findParams = {}
+    if (req.query.lastVote && req.query.lastDate) {
+        findParams = {
+            votes: { $lte: req.query.lastVote },
+            publishedAt: { $lt: req.query.lastDate }
+        }
+    }
+
+    const sortParams = {
+        votes: -1,
+        publishedAt: -1
+    }
+
+    getNextArticles(findParams, sortParams)
+        .then(articles => res.send(articles))
+        .catch((err) => {
+            res.status(500).send({
+                err: `Articles weren't found: ${err}`,
+                reqQuery: req.query
+            })
         })
-        .catch((error) => {
+})
+
+router.get('/new', (req, res) => {
+    let findParams = {}
+    if (req.query.lastDate) {
+        findParams = { publishedAt: { $lt: req.query.lastDate } }
```

Comment:
I read on the docs that $lt works for Date objects, so this will stay this pretty 😄